app/src/main/java/com/shagalalab/sozlik/favorites/FavoritesFragment.java

```diff
@@ -0,0 +1,68 @@
+package com.shagalalab.sozlik.favorites;
+
+
+import android.app.Fragment;
+import android.content.Intent;
+import android.os.Bundle;
+import android.support.v7.widget.LinearLayoutManager;
+import android.support.v7.widget.RecyclerView;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+
+import com.shagalalab.sozlik.R;
+import com.shagalalab.sozlik.model.SozlikDao;
+import com.shagalalab.sozlik.model.SozlikDatabase;
+import com.shagalalab.sozlik.model.SozlikDbModel;
+import com.shagalalab.sozlik.translation.TranslationActivity;
+
+import java.util.List;
+
+/**
+ * A simple {@link Fragment} subclass.
+ */
+public class FavoritesFragment extends Fragment implements FavoritesView, FavoritesAdapter.FavoriteAdapterCallback {
+    private FavoritesPresenter presenter;
+    private FavoritesAdapter adapter;
+    private RecyclerView recyclerView;
+    private SozlikDao sozlikDao;
+    private RecyclerView.LayoutManager layoutManager;
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        View view = inflater.inflate(R.layout.fragment_favorites, container, false);
+        sozlikDao = SozlikDatabase.getSozlikDatabase(getActivity().getApplicationContext()).sozlikDao();
+        presenter = new FavoritesPresenter(this, sozlikDao);
+        recyclerView = view.findViewById(R.id.recycler_favorites);
+        layoutManager = new LinearLayoutManager(view.getContext(), LinearLayoutManager.VERTICAL, false);
+        adapter = new FavoritesAdapter(sozlikDao.getAllFavorites());
```

Comment:
bunı bir qatar joqarıǵa kóter, sóytip tómendegi `adapter.setCallback(this);`ti onıń astına qoy. Ulıwma gruppirovka qılǵan jaqsı, viewlar jaratılǵanı, onnan keyin bos qatar, adapter, bos qatar, recyclerView. Sonda oqiw ańsat boladı